kiosk/src/Components/Background.tsx

```diff
@@ -0,0 +1,100 @@
+import {
+    useEffect,
+    useContext,
+    useMemo,
+    useState,
+    useCallback,
+    useRef,
+} from "react";
+import { AppStateContext } from "../State/AppStateContext";
+import { KioskState } from "../Types";
+
+
+const Background = () => {
+    const { state: kiosk } = useContext(AppStateContext);
+    const [myRef, setMyRef] = useState<HTMLElement | null>(null);
+
+    const canvas = useRef<HTMLCanvasElement>(document.createElement("canvas"));
+    const image = useRef<HTMLImageElement>(document.createElement("img"));
+    const timer = useRef<NodeJS.Timeout | null>(null);
+    // TODO: Store cached background URLs in local storage instead of in memory.
+    const dataUrlCache = useRef<Map<string, string>>(new Map());
+
+    const selectedGame = useMemo(
+        () => kiosk.allGames.find(g => g.id === kiosk.selectedGameId),
+        [kiosk.selectedGameId]
+    );
+
+    const handleRef = useCallback((node: HTMLElement | null) => {
+        setMyRef(node);
+    }, []);
+
+    useEffect(() => {
+        // debounce so we don't update when continuously scrolling games
+        if (timer.current) {
+            clearTimeout(timer.current);
+        }
+
+        if (kiosk.kioskState === KioskState.PlayingGame) {
+            if (myRef) {
+                myRef.style.backgroundImage = "";
+            }
+            return;
+        }
+
+        timer.current = setTimeout(() => {
+            if (selectedGame && kiosk.kioskState !== KioskState.PlayingGame) {
+                if (dataUrlCache.current.has(selectedGame.id)) {
+                    if (myRef) {
+                        myRef.style.backgroundImage = `url(${dataUrlCache.current.get(
+                            selectedGame.id
+                        )})`;
+                    }
+                    return;
+                }
+                image.current.crossOrigin = "anonymous";
+                image.current.src = `https://makecode.com/api/${selectedGame.id}/thumb`;
```

Comment:
I don't see a `thumbnail` member on `GameData`, but I get your meaning. Caching fetched thumbnails is a great idea.

`dataUrlCache` is a little different from a thumbnail cache. It caches a 80x60 rendering of the thumbnail, or the first frame of the thumbnail if it's a gif, as a base-64 encoded png. I have a TODO to apply the acrylic effect to this data rather than having CSS do it. So, `dataUrlCache` is (or will be) a cache of low-res, blurry, dark versions of thumbnails.